lib/src/bitcoin/witness/primed_transaction.rs

```diff
@@ -121,15 +121,14 @@ impl PrimedTransaction {
     pub fn sign_with_rate<C: secp256k1::Signing>(
         self,
         secp: &Secp256k1<C>,
-        fee_per_byte: usize,
+        fee_per_byte: Amount,
     ) -> Result<Transaction, Error> {
         let mut transaction = self._transaction_without_signatures_or_output_values();
 
         let weight = transaction.get_weight();
-        let fee = weight
-            .checked_mul(fee_per_byte)
+        let fee = fee_per_byte
+            .checked_mul(weight as u64)
             .ok_or(Error::OverflowingFee)?;
```

Comment:
Is this actually correct given your recent investigation? Shouldn't we divide the weight by 4 before we multiply it with the fee?

Reply:
`get_weight` is the correct rust-bitoin interface to get the weight of a transaction for fee purposes.

Reply:
To the best of my knowledge, this is correct. We could rename the variable to `fee_per_vByte` to make it clear that it's the virtual size and not the "old" normal tx size.